python/ccxt/async_support/base/ws/future.py

```diff
@@ -45,7 +45,13 @@ def callback(done):
                     if are_all_canceled and future._state == 'PENDING':
                         future.set_exception(ExchangeClosedByUser('Connection closed by the user'))
                         return
+
+                    # handle wait_for scenario
+                    if are_all_canceled and future._state == 'CANCELLED':
```

Comment:
Should this be `if are_all_canceled:` ?

I'm wondering if there can be a case future._state  === FINISHED but are_all_canceled.
In that case `first.result()` would throw an exception